src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -0,0 +1,343 @@
+          encryptionAlgorithm: 'RSA',
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+
+    it('When user is already invited to the workspace, then it should throw', async () => {
+      const invitedUser = newUser();
+      const workspace = newWorkspace();
+      const invitedUserEmail = 'alreadyInvited@example.com';
+      const existingInvite = newWorkspaceInvite({
+        invitedUser: invitedUserEmail,
+      });
+
+      jest
+        .spyOn(workspaceRepository, 'findById')
+        .mockResolvedValueOnce(workspace);
+      jest
+        .spyOn(userUsecases, 'findByEmail')
+        .mockResolvedValueOnce(invitedUser);
+      jest
+        .spyOn(userUsecases, 'findPreCreatedByEmail')
+        .mockResolvedValueOnce(null);
+      jest
+        .spyOn(workspaceRepository, 'findInvite')
+        .mockResolvedValueOnce(existingInvite);
+
+      await expect(
+        service.inviteUserToWorkspace(user, workspace.id, {
+          invitedUser: invitedUserEmail,
+          spaceLimit: BigInt(1024),
+          encryptionKey: 'encryptionKey',
+          encryptionAlgorithm: 'RSA',
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+
+    it('When invitation space limit exceeds the assignable space, then it should throw', async () => {
+      const invitedUser = newUser();
+      const workspace = newWorkspace();
+      const invitedUserEmail = 'newUser@example.com';
+      const spaceLeft = 1024 * 1024 * 10;
+
+      jest
+        .spyOn(workspaceRepository, 'findById')
+        .mockResolvedValueOnce(workspace);
+      jest
+        .spyOn(userUsecases, 'findByEmail')
+        .mockResolvedValueOnce(invitedUser);
+      jest
+        .spyOn(userUsecases, 'findPreCreatedByEmail')
+        .mockResolvedValueOnce(null);
+      jest.spyOn(service, 'isWorkspaceFull').mockResolvedValueOnce(false);
+      jest
+        .spyOn(service, 'getAssignableSpaceInWorkspace')
+        .mockResolvedValueOnce(BigInt(spaceLeft));
+      jest.spyOn(workspaceRepository, 'findInvite').mockResolvedValueOnce(null);
+
+      await expect(
+        service.inviteUserToWorkspace(user, workspace.id, {
+          invitedUser: invitedUserEmail,
+          spaceLimit: BigInt(spaceLeft + 1),
+          encryptionKey: 'encryptionKey',
+          encryptionAlgorithm: 'RSA',
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+  });
+
+  describe('isWorkspaceFull', () => {
+    const workspaceId = 'workspace-id';
+    it('When workspace has slots left, then workspace is not full', async () => {
+      jest
+        .spyOn(workspaceRepository, 'getWorkspaceUsersCount')
+        .mockResolvedValue(5);
+      jest
+        .spyOn(workspaceRepository, 'getWorkspaceInvitationsCount')
+        .mockResolvedValue(4);
+
+      const isFull = await service.isWorkspaceFull(workspaceId);
+      expect(isFull).toBe(false);
+    });
+    it('When workspace has slots left, then workspace is full', async () => {
```

Comment:
I think the legend here is wrong, it should be: 
_When workspace does not have slots, left..._
Instead of: 
_When workspace has slots left_